### ISSUE TITLE
useForm Nested arrays validation not working-  5.0

### ISSUE THREAD

**Issue by kkarwowski** [mantinedev/mantine#1861]
### What package has an issue

@mantine/form

### Describe the bug

When validating nested arrays by `form.validate()` value passed to validate function is undefined:

Form and validation:
```
 const form = useForm({
    initialValues: {
      employees: [{ name: "", active: false, key: randomId() }]
    },
    validate: {
      employees: {
        name: (value) =>
          value.length < 20 ? "Name should have at least 2 letters" : null
      }
    }
```

Error:
` TypeError: value is undefined
`

### In which browser did the problem occur

Fireforx, Chrome

### If possible, please include a link to a codesandbox with the reproduced problem

https://codesandbox.io/s/boring-wing-rrcx4b?file=/src/form.js

### Do you know how to fix the issue

No

### Are you willing to participate in fixing this issue and create a pull request with the fix

No

**Comment by jliit** [comment]
I have a similar problem with validation of nested arrays. Validation for "employees.${index}.name" works but for some reason a validation rule for "employees.name" is added and since that field doesn't exist the value is undefined.

I know that, in my case, value can never be undefined so I did a workaround like this:

`    validate: {
      employees: {
        name: (value) => {
           if (value === undefined)
             return null;
           return value.length < 20 ? "Name should have at least 2 letters" : null
         }
      }
    }`

**Comment by rtivital** [comment]
Fixed in 5.0.2
